examples/Automatic Machine Learning Checkpoint.ipynb

```diff
@@ -0,0 +1,182 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import shutil\n",
+    "import time\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "from neuraxle.base import BaseStep, NonFittableMixin\n",
+    "from neuraxle.checkpoints import DefaultCheckpoint\n",
+    "from neuraxle.hyperparams.distributions import RandInt\n",
+    "from neuraxle.hyperparams.space import HyperparameterSpace\n",
+    "from neuraxle.metaopt.random import RandomSearch\n",
+    "from neuraxle.pipeline import ResumablePipeline, DEFAULT_CACHE_FOLDER, Pipeline\n",
+    "\n",
+    "shutil.rmtree(DEFAULT_CACHE_FOLDER)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Multiplication(NonFittableMixin, BaseStep):\n",
+    "    def __init__(self, sleep_time=0.010, hyperparams=None, hyperparams_space=None):\n",
+    "        BaseStep.__init__(self, hyperparams=hyperparams, hyperparams_space=hyperparams_space)\n",
+    "        self.sleep_time = sleep_time\n",
+    "\n",
+    "    def transform(self, data_inputs):\n",
+    "        time.sleep(self.sleep_time)\n",
+    "        if not isinstance(data_inputs, np.ndarray):\n",
+    "            data_inputs = np.array(data_inputs)\n",
+    "\n",
+    "        return data_inputs * self.hyperparams['hp_mul']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def mean_squared_error(actual_outputs, expected_outputs):\n",
+    "    mses = [(a - b) ** 2 for a, b in zip(actual_outputs, expected_outputs)]\n",
+    "    return sum(mses) / float(len(mses))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_random_search(p, hyperparams_space, data_inputs, expected_outputs):\n",
+    "    time_a = time.time()\n",
+    "    p.set_hyperparams_space(hyperparams_space)\n",
+    "\n",
+    "    random_search = RandomSearch(\n",
+    "        p,\n",
+    "        n_iter=200,\n",
+    "        higher_score_is_better=True,\n",
+    "        print=True\n",
+    "    ).fit(data_inputs, expected_outputs)\n",
+    "\n",
+    "    outputs = random_search.transform(data_inputs)\n",
+    "\n",
+    "    time_b = time.time()\n",
+    "\n",
+    "    actual_score = mean_squared_error(outputs, expected_outputs)\n",
+    "    print('{0} seconds'.format(time_b - time_a))\n",
+    "    print('output: {0}'.format(outputs))\n",
+    "    print('smallest mse: {0}'.format(actual_score))\n",
+    "    print('best hyperparams: {0}'.format(p.get_hyperparams()))\n"
```

Comment:
I would like that you commit the notebook with the outputs printed, instead of a virgin notebook. Before commiting, always do: 
- Reset notebook
- Clear all outputs 
- Rerun all cells 

Those 3 steps above can be found in the file/etc menus above the notebook.

Reply:
This way the reader will actually see the printed times on the documentation site.

Reply:
It doesnt work, jupyter is not using my local version of neuraxle :(

Reply:
@alexbrillant You need to add the virtual environment to the notebook

https://anbasile.github.io/programming/2017/06/25/jupyter-venv/

https://medium.com/@eleroy/jupyter-notebook-in-a-virtual-environment-virtualenv-8f3c3448247

Then choose the good kernel type in the jupyter web UI interface.